polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala

```diff
@@ -198,28 +230,33 @@ class PythonInterpreter(val kernelContext: KernelContext) extends LanguageInterp
         jep.set("__polynote_globals__", globals)
         expandGlobals()
 
-        jep.eval("__polynote_parsed__ = ast.parse(__polynote_code__, __polynote_cell__, 'exec')\n")
-
-        val numStats = jep.getValue("len(__polynote_parsed__.body)", classOf[java.lang.Long])
-
-        if (numStats > 0) {
-          jep.eval("__polynote_parsed__ = LastExprAssigner().visit(__polynote_parsed__)")
-          jep.eval("__polynote_parsed__ = ast.fix_missing_locations(__polynote_parsed__)")
-          jep.eval("__polynote_code__ = compile(__polynote_parsed__, '<ast>', 'exec')")
 
+        val pyResults = for {
+          _ <- wrapEval("__polynote_parsed__ = ast.parse(__polynote_code__, __polynote_cell__, 'exec')").right
+          _ <- (if (jep.getValue("len(__polynote_parsed__.body)", classOf[java.lang.Integer]) > 0) Right(List.empty[Result]) else Left(List.empty[CompileErrors])).right
```

Comment:
😬

Reply:
If you import `cats.syntax.either._` then you don't need to say `.right` anymore

Reply:
Done, thanks for the tip!